Handler/duration.ts

```diff
@@ -7,40 +7,60 @@ import { add } from "./base"
 
 class Handler implements Converter<{ hours: number; minutes: number } | undefined>, Formatter {
 	private pattern: RegExp
-	constructor(readonly suffix: string = "h") {
-		const pattern = this.suffix
-			.split("")
-			.map(symbol => symbol + "{0,1}")
-			.join("")
-		this.pattern = new RegExp(`^\\d*:{0,1}[0-5]{0,1}[0-9]{0,1}${pattern}$`)
+	private decimal: boolean
+	constructor() {
+		this.pattern = new RegExp(`^\\d*:{0,1}[0-5]{0,1}[0-9]{0,1}$|^\\d*\\.{0,1}[0-9]{0,2}$`)
+		// normal time												|		decimal time
+		// ^\\d*:{0,1}[0-5]{0,1}[0-9]{0,1}$   |   ^\\d*\\.{0,1}[0-9]{0,2}$
 	}
 	toString(data: { hours: number; minutes: number } | undefined): string {
 		return `${data?.hours?.toString(10) ?? "0"}:${data?.minutes?.toString(10).padStart(2, "0") ?? "00"}`
 	}
 	fromString(value: string): { hours: number; minutes: number } | undefined {
-		const splitted = typeof value == "string" && value.split(":", 2).map(value => Number.parseInt(value))
-		return splitted
-			? {
-					hours: !splitted[0] || !Number.isFinite(splitted[0]) ? 0 : splitted[0],
-					minutes: !splitted[1] || !Number.isFinite(splitted[1]) ? 0 : splitted[1],
-			  }
-			: undefined
+		let result: undefined | { hours: number; minutes: number }
+		if (!this.decimal) {
+			const splitted = typeof value == "string" && value.split(":", 2).map(value => Number.parseInt(value))
+			result = splitted
+				? {
+						hours: !splitted[0] || !Number.isFinite(splitted[0]) ? 0 : splitted[0],
+						minutes: !splitted[1] || !Number.isFinite(splitted[1]) ? 0 : splitted[1],
+				  }
+				: undefined
+		} else if (this.decimal) {
+			const splittedString = typeof value == "string" && value.split(".", 2)
+			if (splittedString) {
+				splittedString[1] = splittedString[1].length == 1 ? splittedString[1] + "0" : splittedString[1]
+			}
+			const splitted = splittedString ? splittedString.map(value => Number.parseInt(value)) : undefined
+			result = splitted
+				? {
+						hours: !splitted[0] || !Number.isFinite(splitted[0]) ? 0 : splitted[0],
+						minutes: !splitted[1] || splitted[1] < 100 ? (splitted[1] / 100) * 60 : splitted[1],
```

Comment:
`!splitted[1] || splitted[1] < 100 ? (splitted[1] / 100) * 60 : splitted[1]`

The function should never return ` splitted[1]` but the fact remains that if `splitted[1] < 100 ` is false then it could. It should never be equal to or greater than a 100 but if it is then this function handles it wrongly.